### PR TITLE
[improvement] Include MusicVideos when creating playlists

### PR DESCRIPTION
**Pull request by VioletRed**
Show videodb://musicvideos/ as a source for items in playlists.

Right now only songs can be added into playlists. This patch allows adding music videos as well.

### CODE REVIEW

**Comment by VioletRed** [comment]
Anyone checking this?

File: xbmc/music/windows/GUIWindowMusicBase.cpp [hunk 1]
```diff
@@ -706,7 +706,10 @@ void CGUIWindowMusicBase::UpdateButtons()
   msg2.SetLabel(g_localizeStrings.Get(744)); // Files
   g_windowManager.SendMessage(msg2);
 
-  msg2.SetLabel(g_localizeStrings.Get(15100)); // Library
+  msg2.SetLabel(g_localizeStrings.Get(14022)); // Library
```
Comment (Montellese) [inline xbmc/music/windows/GUIWindowMusicBase.cpp]: Any reason you changed the label here? Both say "Library" but one is used here and one in settings.

File: xbmc/music/windows/GUIWindowMusicPlaylistEditor.cpp [hunk 2]
```diff
@@ -155,6 +155,12 @@ bool CGUIWindowMusicPlaylistEditor::GetDirectory(const std::string &strDirectory
     db->m_bIsShareOrDrive = true;
     items.SetPath("");
     items.Add(db);
+    CFileItemPtr vdb(new CFileItem("videodb://musicvideos/", true));
```
Comment (Montellese) [inline xbmc/music/windows/GUIWindowMusicPlaylistEditor.cpp]: Please add an empty line (and maybe also one in the existing code between "files://" and "musicdb://") for better readability.

File: xbmc/video/windows/GUIWindowVideoBase.cpp [hunk 3]
```diff
@@ -1635,11 +1635,11 @@ bool CGUIWindowVideoBase::GetDirectory(const std::string &strDirectory, CFileIte
     newPlaylist->m_bIsFolder = true;
     items.Add(newPlaylist);
 
-/*    newPlaylist.reset(new CFileItem("newplaylist://", false));
```
Comment (Montellese) [inline xbmc/video/windows/GUIWindowVideoBase.cpp]: Are you sure that the playlist implementation works properly now? Either way IMO this should go into a separate commit and PR.

File: xbmc/music/windows/GUIWindowMusicBase.cpp (hunk 1 above)
Comment (VioletRed) [inline xbmc/music/windows/GUIWindowMusicBase.cpp]: Because they both refer to the same menu item. It's created with label 14022, but then refreshed with label 15100. Seems just wrong.

File: xbmc/music/windows/GUIWindowMusicPlaylistEditor.cpp (hunk 2 above)
Comment (VioletRed) [inline xbmc/music/windows/GUIWindowMusicPlaylistEditor.cpp]: Fixed.

File: xbmc/video/windows/GUIWindowVideoBase.cpp (hunk 3 above)
Comment (VioletRed) [inline xbmc/video/windows/GUIWindowVideoBase.cpp]: It's working now. The prerequisites for this change to work are already merged in master.

I'll split it in two commits, but why should they be split in different PR? Adding the music video library to the music menu, but not the music library sounds like a half-done feature.

Comment (Montellese) [inline xbmc/video/windows/GUIWindowVideoBase.cpp]: Well one is touching the music library and the other the video library. And the "newplaylist://" can not only be used for musicvideos IIRC so this is not strictly a music video only change.

**Comment by mkortstiege** [comment]
For some reason all those playlist code touching PRs makes me shudder ;) Simplest changes can trigger a lot of regressions ..

File: xbmc/music/windows/GUIWindowMusicBase.cpp (hunk 1 above)
Comment (Montellese) [inline xbmc/music/windows/GUIWindowMusicBase.cpp]: Then please add a line to the `strings.po` file that the label is used in this file.

**Comment by Montellese** [comment]
@mkortstiege: That's why I asked if he was sure that video playlist are working properly.

File: xbmc/video/windows/GUIWindowVideoBase.cpp (hunk 3 above)
Comment (VioletRed) [inline xbmc/video/windows/GUIWindowVideoBase.cpp]: Makes sense. Done.

**Comment by VioletRed** [comment]
@mkortstiege Example of previous problems? I would give me an idea about what the weakest points of the implementation are.

File: xbmc/music/windows/GUIWindowMusicBase.cpp (hunk 1 above)
Comment (VioletRed) [inline xbmc/music/windows/GUIWindowMusicBase.cpp]: Done.

Looking at the code, 14022 is used in GUIWindowMusicPlaylistEditor.cpp, GUIWindowVideoBase.cpp, and 15100 was used only here. But strings.po says that 14022 is used in settings, and says nothing about 15100. 

Both messages have been in strings.po from the very beginning.Any idea about why?

**Comment by mkortstiege** [comment]
Hard to tell. The entire code is hard to read and to maintain which actually makes it so different to change something without raising regressions.

**Comment by VioletRed** [comment]
The only regression I've seen so far is not related to my changes, and is that "Artists" view in the video database doesn't show anything.

**Comment by VioletRed** [comment]
Pinging on this.

**Comment by MartijnKaijser** [comment]
@Montellese any changes needed or good to go?

**Comment by Montellese** [comment]
Sorry forgot about it. jenkins build and merge
